packages/frontend/src/app/api/(public)/scaling/activity/[slug]/route.ts

```diff
@@ -0,0 +1,67 @@
+import { layer2s, layer3s } from '@l2beat/config'
+import { ProjectId } from '@l2beat/shared-pure'
+import { type NextRequest, NextResponse } from 'next/server'
+import { getActivityChart } from '~/server/features/scaling/activity/get-activity-chart'
+import { type ActivityProjectFilter } from '~/server/features/scaling/activity/utils/project-filter-utils'
+import { ActivityTimeRange } from '~/server/features/scaling/activity/utils/range'
+
+const projectsIds = [...layer2s, ...layer3s]
+  .map((p) => ({
+    id: p.id,
+    slug: p.display.slug,
+  }))
+  .concat({
+    id: ProjectId.ETHEREUM,
+    slug: 'ethereum',
+  })
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: { slug: string } },
+) {
+  const searchParams = request.nextUrl.searchParams
+  const range = ActivityTimeRange.catch('30d').parse(searchParams.get('range'))
+
+  const project = projectsIds.find((p) => p.slug === params.slug)
+
+  if (!project) {
+    return NextResponse.json({
+      success: false,
+      error: 'Project not found.',
+    })
+  }
+
+  const isEthereum = project.slug === 'ethereum'
+
+  const filter: ActivityProjectFilter = isEthereum
+    ? { type: 'all' }
+    : { type: 'projects', projectIds: [project.id] }
```

Comment:
we want to show all project if ethereum is passed as slug?

Reply:
If we could use ethereum as ProjectId then I would have used it - but that's no the case - we pull 'something' and extract ethereum data points - look below

Reply:
got it